src/Sarif/Visitors/RemapIndicesVisitor.cs

```diff
@@ -55,17 +55,17 @@ public override Result VisitResult(Result node)
 
         public override Location VisitLocation(Location node)
         {
-            if (node.LogicalLocationIndex != -1 && HistoricalLogicalLocations != null)
+            if (node.LogicalLocation != null && node.LogicalLocation.ParentIndex != -1 && HistoricalLogicalLocations != null)
             {
-                LogicalLocation logicalLocation = HistoricalLogicalLocations[node.LogicalLocationIndex];
+                LogicalLocation logicalLocation = HistoricalLogicalLocations[node.LogicalLocation.ParentIndex];
```

Comment:
>ParentIndex [](start = 98, length = 11)

so this is wrong, you need to add an 'index' property to logical location. the parent is used to point to the parent, not the actual logical location index.